src/ui/StarRating/StarRating.ts

```diff
@@ -0,0 +1,126 @@
+import 'styling/_StarRating';
+import { $$ } from '../../utils/Dom';
+import { Utils } from '../../utils/Utils';
+import { l } from '../../strings/Strings';
+import { Component } from '../Base/Component';
+import { SVGIcons } from '../../utils/SVGIcons';
+import { exportGlobally } from '../../GlobalExports';
+import { IQueryResult } from '../../rest/QueryResult';
+import { IComponentBindings } from '../Base/ComponentBindings';
+import { ComponentOptions, IFieldOption } from '../Base/ComponentOptions';
+import { Initialization } from '../Base/Initialization';
+
+const DEFAULT_SCALE = 5;
+
+export interface IStarRatingOptions {
+  rating?: IFieldOption;
+  numberOfRatings?: IFieldOption;
+  ratingScale?: number;
+}
+/**
+ * The `StarRating` component renders a 5-star rating widget.
+ *
+ * This component is a result template component (see [Result Templates](https://developers.coveo.com/x/aIGfAQ)).
+ */
+export class StarRating extends Component {
+  static ID = 'StarRating';
+  private rating: number;
+  private numberOfRatings: number;
+  private ratingScale: number;
+
+  static doExport = () => {
+    exportGlobally({
+      StarRating: StarRating
+    });
+  };
+
+  static options: IStarRatingOptions = {
+    /**
+     * Specifies the rating to be displayed as stars.
+     *
+     * Specifying a value for this parameter is required in order for the StarRating component to work.
+     */
+    rating: ComponentOptions.buildFieldOption({ defaultValue: '@rating', required: true }),
+
+    /**
+     * Specifies the number to be displayed in the label indicating the number of ratings.
+     *
+     * If unspecified, no ratings number label is shown. If a value of `0` is provided, a `No Ratings` label is displayed
+     */
+    numberOfRatings: ComponentOptions.buildFieldOption({ required: false }),
+
+    /**
+     * Specifies the scale on which ratings are to be applied
+     *
+     * Default value is `5`
+     */
+    ratingScale: ComponentOptions.buildNumberOption({ defaultValue: DEFAULT_SCALE })
+  };
+
+  /**
+   * Creates a new `StarRating` component.
+   * @param element The HTMLElement on which to instantiate the component.
+   * @param options The options for the `StarRating` component.
+   * @param bindings The bindings that the component requires to function normally.
+   */
+  constructor(
+    public element: HTMLElement,
+    public options: IStarRatingOptions,
+    public bindings?: IComponentBindings,
+    result?: IQueryResult
+  ) {
+    super(element, StarRating.ID);
+    this.options = ComponentOptions.initComponentOptions(element, StarRating, options);
+
+    if (result) {
+      if (this.getValuesFromFields(result)) {
+        this.renderComponent(element);
+      }
+    } else if (this.element.parentElement != null) {
+      this.element.parentElement.removeChild(this.element);
```

Comment:
I would not remove the component from the DOM. Just leave it there, but with with a constructor that essentially do nothing.

Add a warning/error message using `this.logger` that explains the problem.

Reply:
What would be the advantage of converting it to a getter?
I agree with moving the checks to the render function 👍

Reply:
Getters are easier for other developers to use because they remove the need to understand what to pass and where to get it from.